tf2onnx/optimizer/merge_duplicated_nodes_optimizer.py

```diff
@@ -90,3 +92,41 @@ def _skip_node_type(node):
     @staticmethod
     def _len_of_node_output(node):
         return len(node.output)
+
+    @staticmethod
+    def _are_attr_equal(a, b):
+        if len(a) != len(b):
+            return False
+        for k, v in a.items():
+            if v.type == onnx_pb.AttributeProto.FLOAT or \
+                    v.type == onnx_pb.AttributeProto.INT or \
+                    v.type == onnx_pb.AttributeProto.STRING or \
+                    v.type == onnx_pb.AttributeProto.FLOATS or \
+                    v.type == onnx_pb.AttributeProto.INTS or \
+                    v.type == onnx_pb.AttributeProto.STRINGS:
+                if v != b[k]:
+                    return False
+            elif v.type == onnx_pb.AttributeProto.TENSOR:
+                t1 = v.t
+                t2 = b[k].t
+                if not (t1.dims == t2.dims and t1.data_type == t2.data_type and t1.raw_data == t2.raw_data):
```

Comment:
what if t1 stores data in data_type while t2 stores data in raw_data

Reply:
raw_data must almost always have the data regardless of its type. Please let me know if I'm wrong in understanding this.

Reply:
i'm afrad not, tensor can decide their data in data_type field or raw_data.

Reply:
`make_tensor` has the option `raw`.

Reply:
I'm going to investigate this a little further and make necessary changes

Reply:
The current comparison looks fine to me.
It's too difficult to compare raw data with the plain number, as we barely know which form the raw data comes from. It can be from list, numpy, string, etc..